e2e/libs/node/node.py

```diff
@@ -1,26 +1,42 @@
 from kubernetes import client
+import yaml
+import time
+import logging
+from utility.utility import apply_cr_from_yaml, get_cr
 
-class Node:
-
-    _instance = None
-
-    def __new__(cls, *args, **kwargs):
-        if cls._instance is None:
-            cls._instance = super().__new__(cls)
-        return cls._instance
+RETRY_COUNT = 180
+RETRY_INTERVAL = 1
 
-    def __init__(self):
-        self.core_api = client.CoreV1Api()
-
-    def get(self):
-        obj = self.core_api.list_node()
-        nodes = []
-        for item in obj.items:
-            if 'node-role.kubernetes.io/control-plane' not in item.metadata.labels and \
-                    'node-role.kubernetes.io/master' not in item.metadata.labels:
-                nodes.append(item.metadata.name)
-        return sorted(nodes)
+class Node:
 
-    def get_by_index(self, index):
-        nodes = self.get()
-        return nodes[index]
\ No newline at end of file
+    def reboot_node(self, running_on_node_name, reboot_node_name, shut_down_time_in_sec=10):
+        with open('/tmp/instance_mapping', 'r') as f:
```

Comment:
Please use empty line to separate different logic operations. This will make code clear.

Also, just use log instead of a comment if it's suitable. 

```python
       with open('/tmp/instance_mapping', 'r') as f:
            mapping = yaml.safe_load(f)
            reboot_node_id = mapping[reboot_node_name]
            
        filepath = './litmus/reboot-engine.yaml'
        with open(filepath, 'r') as f:
            data = yaml.safe_load(f)
            data['spec']['components']['runner']['nodeSelector']['kubernetes.io/hostname'] = running_on_node_name
            data['spec']['experiments'][0]['spec']['components']['nodeSelector']['kubernetes.io/hostname'] = running_on_node_name
            data['spec']['experiments'][0]['spec']['components']['env'][1]['value'] = str(shut_down_time_in_sec)
            data['spec']['experiments'][0]['spec']['components']['env'][2]['value'] = reboot_node_id
            
        with open(filepath, 'w') as file:
            yaml.dump(data,file,sort_keys=False)
            
        apply_cr_from_yaml(filepath)
```

Reply:
Refined.